internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java

```diff
@@ -0,0 +1,328 @@
+                Path unifiedFilePath = Paths.get(scanDirectoryLocation.toPath().toString(), unifiedFileName);
+                writeToFile(entry.getValue(), unifiedFilePath);
+                storedFileMap.put(entry.getKey(), unifiedFilePath.toString());
+            }
+
+            Iterator<Map.Entry<String, String>> iter = parameterMap.entrySet().iterator();
+            while (iter.hasNext()) {
+                Map.Entry<String, String> param = iter.next();
+                if (param.getValue().isEmpty()) {
+                    iter.remove();
+                } else {
+                    if (param.getKey().contains("@byURL")) {
+
+                        // if the param map contains files as a URL (param names containing "@byURL"), add them to
+                        // a separate map to download later.
+                        String fileExtension = FilenameUtils.getExtension(param.getValue());
+                        if (StringUtils.isBlank(fileExtension)) {
+                            throw new ScanManagerWebappException("Unable to identify the file extension from the " +
+                                    "file URL: " + param.getValue());
+                        }
+                        String fileName =
+                                StringUtils.substringBeforeLast(param.getKey(), "@") + "." + fileExtension;
+
+                        if (!filesToBeDownloadedFromURL.containsKey(fileName)) {
+                            filesToBeDownloadedFromURL.put(fileName, param.getValue());
+                        }
+                    }
+                }
+            }
+
+            // begin the pre scans tasks and initiate the scan submission.
+            new Thread(() -> beginScanSubmit(storedFileMap, parameterMap, filesToBeDownloadedFromURL, scanDirectory,
+                    scanWaiting), "BeginScanSubmitToScanManagerAPI").start();
+        } catch (ScanManagerWebappException e) {
+            logService.insertError(scanWaiting, e);
+
+            // Delete local scan artifacts directory.
+            try {
+                FileUtils.deleteDirectory(scanDirectoryLocation);
+            } catch (IOException ex) {
+                logService.insertError(scanWaiting, ex);
+            }
+        }
+        return scanWaiting;
+    }
+
+    private void writeToFile(MultipartFile file, Path filepath) throws ScanManagerWebappException {
+        try (OutputStream os = Files.newOutputStream(filepath)) {
+            os.write(file.getBytes());
+        } catch (IOException e) {
+            throw new ScanManagerWebappException("IO Exception occurred while writing files to the disk", e);
+        }
+    }
+
+    private Scan addScanToWaiting(Map<String, String> parameterMap) {
+        String preJobId = PRE_JOB_ID_PREFIX + UUID.randomUUID().toString();
+        Scan scanWaiting = new Scan(preJobId);
+        scanWaiting.setName(parameterMap.get(SCAN_REQUEST_SCAN_NAME_ATTRIBUTE_NAME));
```

Comment:
Usage of `waitingScans` and `scanWaiting` is bit odd. Can we call this `waitingScan`?

Last statement will be ` waitingScans.put(preJobId, waitingScan);`